cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -48,12 +50,17 @@ type autoScaling interface {
 	TerminateInstanceInAutoScalingGroup(input *autoscaling.TerminateInstanceInAutoScalingGroupInput) (*autoscaling.TerminateInstanceInAutoScalingGroupOutput, error)
 }
 
+type kubeEC2 interface {
+	DescribeInstances(input *ec2.DescribeInstancesInput) (*ec2.DescribeInstancesOutput, error)
+}
+
 // AwsManager is handles aws communication and data caching.
 type AwsManager struct {
 	asgs     []*asgInformation
 	asgCache map[AwsRef]*Asg
 
 	service    autoScaling
+	ec2Service kubeEC2
```

Comment:
Seems like a bit of a code smell that we named this service `ec2Service` but the autoscaling service is just named `service`